controller/src/main/java/org/jboss/as/controller/AttributeDefinition.java

```diff
@@ -290,11 +292,24 @@ public ModelNode getDefaultValue() {
      *
      * @return the name of the group, or {@code null} if the attribute is not associated with a group
      */
-    public String getAttributeGroup() {
+    public AttributeGroup getAttributeGroup() {
         return attributeGroup;
     }
 
     /**
+     * Gets the stringified name of the attribute group with which this attribute is associated, if any.
+     *
+     * @return the stringified name of the group, or {@code null} if the attribute is not associated with a group
+     */
+    public String getStringifiedAttributeGroup() {
```

Comment:
I don't like this here; it's bloat to the already large AttributeDefinition API just to save a null check in one place.